playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -229,3 +229,56 @@
   tasks:
   - include: docker/upgrade.yml
     when: l_docker_upgrade is defined and l_docker_upgrade | bool and not openshift.common.is_atomic | bool
+
+- name: Drain and upgrade master nodes
```

Comment:
That will not help. Otherwise there is a way to postpone the restart phase off both master and node upgrades into a single point in time. Optionally, we can skip the restart of openswitch, docker and node service. But then, we would have to do that in the master upgrade play.